src/api/controllers/FeedbackController.ts

```diff
@@ -0,0 +1,71 @@
+import { Body, Delete, Get, JsonController, Params, Post } from 'routing-controllers';
+
+import { FeedbackService } from '../../services/FeedbackService';
+import { CreateFeedbackRequest, ErrorResponse, getErrorMessage, GetFeedbackResponse, GetFeedbacksResponse, GetSearchedFeedbackRequest } from '../../types';
+import { UuidParam } from '../validators/GenericRequests';
+
+@JsonController('feedback/')
+export class FeedbackController {
+  private feedbackService: FeedbackService;
+
+  constructor(feedbackService: FeedbackService) {
+    this.feedbackService = feedbackService;
+  }
+
+  @Get()
+  async getAllFeedback(): Promise<GetFeedbacksResponse | ErrorResponse> {
+    try {
+      const feedback = await this.feedbackService.getAllFeedback();
+      return { feedback: feedback.map((f) => f.getFeedbackInfo()) };
+    } catch (error) {
+      return { error: getErrorMessage(error) }
+    }
+  }
+
+  @Get('id/:id/')
+  async getFeedbackById(@Params() params: UuidParam): Promise<GetFeedbackResponse | ErrorResponse> {
+    try {
+      return { feedback: await this.feedbackService.getFeedbackById(params.id) };
+    } catch (error) {
+      return { error: getErrorMessage(error) }
+    }
+  }
+
+  @Get('userId/:id/')
+  async getFeedbackByUserId(@Params() params: UuidParam): Promise<GetFeedbacksResponse | ErrorResponse> {
+    try {
+      return { feedback: await this.feedbackService.getFeedbackByUserId(params.id) };
+    } catch (error) {
+      return { error: getErrorMessage(error) }
+    }
+  }
+
+  @Post()
+  async createPost(@Body() createFeedbackRequest: CreateFeedbackRequest): Promise<GetFeedbackResponse | ErrorResponse> {
```

Comment:
Maybe rename this function to `createFeedback`? I know we never call controller functions but a little confusing since we have a Posts model